src/status_im/subs/wallet/collectibles.cljs

```diff
@@ -91,35 +90,36 @@
              current-account-collectibles))))
 
 (re-frame/reg-sub
- :wallet/last-collectible-details
- :<- [:wallet]
- (fn [wallet]
-   (let [last-collectible (:last-collectible-details wallet)]
-     (assoc last-collectible :preview-url (preview-url last-collectible)))))
+ :wallet/collectible
+ :<- [:wallet/ui]
+ :-> :collectible)
 
 (re-frame/reg-sub
- :wallet/last-collectible-aspect-ratio
- :<- [:wallet]
- (fn [wallet]
-   (:last-collectible-aspect-ratio wallet)))
+ :wallet/collectible-details
+ :<- [:wallet/collectible]
+ (fn [collectible]
+   (as-> collectible $
+     (:details $)
+     (assoc $ :preview-url (preview-url $)))))
 
 (re-frame/reg-sub
- :wallet/last-collectible-details-chain-id
- :<- [:wallet/last-collectible-details]
+ :wallet/collectible-aspect-ratio
+ :<- [:wallet/collectible]
  (fn [collectible]
-   (get-in collectible [:id :contract-id :chain-id])))
+   (:aspect-ratio collectible 1)))
 
 (re-frame/reg-sub
- :wallet/last-collectible-details-traits
- :<- [:wallet/last-collectible-details]
+ :wallet/collectible-gradient-color
+ :<- [:wallet/collectible]
  (fn [collectible]
-   (get-in collectible [:collectible-data :traits])))
+   (:gradient-color collectible :gradient-1)))
 
 (re-frame/reg-sub
- :wallet/last-collectible-details-owner
- :<- [:wallet/last-collectible-details]
- :<- [:wallet]
- (fn [[collectible wallet]]
-   (let [address (:address (first (:ownership collectible)))
-         account (get-in wallet [:accounts address])]
-     account)))
+ :wallet/collectible-details-owner
+ :<- [:wallet/accounts]
+ (fn [accounts [_ collectible]]
+   (let [collectible-address (-> collectible :ownership first :address)]
+     (some #(when (= (:address %) collectible-address)
+              %)
+           accounts))))
```

Comment:
Updated this sub to receive the collectible we want to use, this change makes this sub reusable

Reply:
Also it's been modified to depend on `wallet/accounts` instead of just the `:wallet` (which is updated very often)